src/twisted/logger/_filter.py

```diff
@@ -176,44 +171,41 @@ def logLevelForNamespace(self, namespace):
                 return self._logLevelsByNamespace[namespace]
             index -= 1
 
-        return self._logLevelsByNamespace[None]
+        return self._logLevelsByNamespace[""]
 
-    def setLogLevelForNamespace(self, namespace, level):
+    def setLogLevelForNamespace(self, namespace: str, level: NamedConstant) -> None:
         """
         Sets the log level for a logging namespace.
 
         @param namespace: A logging namespace.
-        @type namespace: L{str} (native string)
-
         @param level: The log level for the given namespace.
-        @type level: L{LogLevel}
         """
         if level not in LogLevel.iterconstants():
             raise InvalidLogLevelError(level)
 
         if namespace:
             self._logLevelsByNamespace[namespace] = level
         else:
-            self._logLevelsByNamespace[None] = level
+            self._logLevelsByNamespace[""] = level
 
-    def clearLogLevels(self):
+    def clearLogLevels(self) -> None:
         """
         Clears all log levels to the default.
         """
         self._logLevelsByNamespace.clear()
-        self._logLevelsByNamespace[None] = self.defaultLogLevel
+        self._logLevelsByNamespace[""] = self.defaultLogLevel
 
-    def __call__(self, event):
+    def __call__(self, event: LogEvent) -> NamedConstant:
         eventLevel = event.get("log_level", None)
-        namespace = event.get("log_namespace", None)
-        namespaceLevel = self.logLevelForNamespace(namespace)
+        if eventLevel is None:
+            return PredicateResult.no
 
-        if (
-            eventLevel is None
-            or namespace is None
-            or LogLevel._priorityForLevel(eventLevel)
-            < LogLevel._priorityForLevel(namespaceLevel)
```

Comment:
We get rid of `_priorityForLevel` in this PR. It's no longer needed because [Trac ticket #6523](https://twistedmatrix.com/trac/ticket/6523) is fixed, so comparing the levels directly works.